architecture-model-generator/modules/architecture-model-generator-core/src/main/java/io/ballerina/artifactsgenerator/ModuleNodeTransformer.java

```diff
@@ -119,9 +119,11 @@ public Optional<Artifact> transform(ServiceDeclarationNode serviceDeclarationNod
         }
 
         // Generate the service path
-        String absolutePath = getPathString(serviceDeclarationNode.absoluteResourcePath());
+        String entryPointName = serviceDeclarationNode.typeDescriptor()
```

Comment:
If typeDescriptor or resourcePath is not present we have to use Listener to resolve the service type.

Reply:
do u have an example library for this?

Reply:
since the typeDescriptor is optional we have to resolve the service type using the attached listener.

Refer this [kafka service bbe](https://ballerina.io/learn/by-example/kafka-service-consume-message/).